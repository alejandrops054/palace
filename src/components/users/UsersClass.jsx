import React from 'react'
import { Button, Row, Input, Label, Col, Form, FormGroup } from 'reactstrap'
import Table from "rc-table";




const UsersClass = () =>{
    const columns = [
        {
            title: 'Code',
            dataIndex: 'Code',
            key: '1',
            //    render: () => <Checkbox  value label col/>,
            width: 180,
            align: 'center'
        }, {
            title: 'Description',
            dataIndex: 'Description',
            key: '2',
            width: 180,
            align: 'center'
        }, {
            title: 'System',
            dataIndex: 'System',
            key: '3',
            width: 180,
            align: 'center'
        }
    ]

    const data = [
        {
            Code: 'CARTERA-CBK',
            Description: 'CARTERA CHANGE BANKS',
            System:'CARTERA OF CHANGES BANKS',
        }, 
        
    ];

    return(
        <>
            <h1>Users Class</h1>
            <Form>
                <Row>
                    <Col md={6}>
                        <Label>System</Label>
                        <Input type='select' style={{width:'60%'}}/>
                    </Col>
                    <Col md={6} style={{paddingTop:'2%'}}>
                        <Button color="primary">Apply filtler</Button>
                    </Col>
                </Row>
                <Row>
                    <Col md={6}>
                        <Label>Code</Label>
                        <Input type='select' style={{width:'60%'}}/>
                    </Col>
                    <Col md={6} style={{paddingTop:'2%'}}>
                        <Button color="primary">Clear filtler</Button>
                    </Col>
                </Row>
                <Label>Description</Label>
                <Input type='text' style={{width:'60%'}}/>
                <Label>Override</Label>
                <Input type='select' style={{width:'60%'}}/>
                <FormGroup>
                <Table className="table" id="resume-table"
                        columns={columns}
                        data={data}/>
                </FormGroup>
            </Form>
        </>
    )
}

export default UsersClass