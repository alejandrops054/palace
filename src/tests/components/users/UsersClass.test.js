/* Sistema Intelligence 2.0
Módulo Contratos
Versión 1.0
Author: IntregraIT Contratos
Comentarios: xxxxx
*/
import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import UsersClass from '../../../components/users/UsersClass'

describe('UsersClass component', () => {
    test('debe mostrar snapshot', () => {

        const wrapper = shallow(<UsersClass />)
        expect(wrapper).toMatchSnapshot();
    })

})