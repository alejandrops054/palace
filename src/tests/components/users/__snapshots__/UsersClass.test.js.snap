// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersClass component debe mostrar snapshot 1`] = `
<Fragment>
  <h1>
    Users Class
  </h1>
  <Form
    tag="form"
  >
    <Row
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Col
        md={6}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          System
        </Label>
        <Input
          style={
            Object {
              "width": "60%",
            }
          }
          type="select"
        />
      </Col>
      <Col
        md={6}
        style={
          Object {
            "paddingTop": "2%",
          }
        }
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Button
          color="primary"
          tag="button"
        >
          Apply filtler
        </Button>
      </Col>
    </Row>
    <Row
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Col
        md={6}
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Label
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Code
        </Label>
        <Input
          style={
            Object {
              "width": "60%",
            }
          }
          type="select"
        />
      </Col>
      <Col
        md={6}
        style={
          Object {
            "paddingTop": "2%",
          }
        }
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <Button
          color="primary"
          tag="button"
        >
          Clear filtler
        </Button>
      </Col>
    </Row>
    <Label
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Description
    </Label>
    <Input
      style={
        Object {
          "width": "60%",
        }
      }
      type="text"
    />
    <Label
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Override
    </Label>
    <Input
      style={
        Object {
          "width": "60%",
        }
      }
      type="select"
    />
    <FormGroup
      tag="div"
    >
      <Table
        className="table"
        columns={
          Array [
            Object {
              "align": "center",
              "dataIndex": "Code",
              "key": "1",
              "title": "Code",
              "width": 180,
            },
            Object {
              "align": "center",
              "dataIndex": "Description",
              "key": "2",
              "title": "Description",
              "width": 180,
            },
            Object {
              "align": "center",
              "dataIndex": "System",
              "key": "3",
              "title": "System",
              "width": 180,
            },
          ]
        }
        data={
          Array [
            Object {
              "Code": "CARTERA-CBK",
              "Description": "CARTERA CHANGE BANKS",
              "System": "CARTERA OF CHANGES BANKS",
            },
          ]
        }
        emptyText={[Function]}
        id="resume-table"
        prefixCls="rc-table"
        rowKey="key"
      />
    </FormGroup>
  </Form>
</Fragment>
`;
